### ISSUE TITLE
Update to the latest code

### ISSUE THREAD

**Issue by toddparker** [filamentgroup/select-css#13]
### Latest version:

http://jsbin.com/horebic/
### What's new:
- Wraps the display of the custom arrow and `appearance: none` behind `@supports`
- Moves all hacks into an optional section
- Leverages FF 35 support for `-moz-appearance`
- Ensures that IE 8/9 doesn't show the custom arrow and native select arrow (old bug)
- IE 10/11 has a custom arrow since we can hide the native arrow via hackiness
- Removes the Opera hack for double arrows since we now shield the styles via `@support`
### Bug: Removing the FF hacks shows a double arrow in v4-35

For FF, if I remove our hack to make it look good in versions 4-35, it looks like it's still showing the custom arrow even though it's `display: none` and visibility is tucked behind `@supports`. I had to add this to hide the arrow - why would older FF be passing `@supports ( -moz-appearance: none )` to show the arrow?

For now, I've added this rule inside `@supports`, then display it inside the old FF hack so we can delete the FF hacks and the native arrow will show up. Can we get rid of this?

```
    /* Show only the native arrow */
    @-moz-document url-prefix() { 
      .custom-select::after {
        display:none;
      }
    }
```
### I've confirmed I can remove the IE hacks and it falls back nicely to the native arrow with custom button styling

TODO:
- Separate out demo styles from core select styles
- Break out backcompat for IE and FF - TODO: make sure we fallback to a native select arrow if these aren't in play
- Final testing
- Upstream into formcore

**Comment by karlcow** [comment]
Currently we get

![capture d ecran 2016-02-26 a 10 02 50](https://cloud.githubusercontent.com/assets/505230/13340003/25d7dce2-dc70-11e5-8361-3c91eb06dc8d.png)

Should this be

```css
.custom-select select {
	-moz-appearance: none;
	width: calc(100% - 1rem);
	padding-right: 1em;
}
```

to get this result: 

![capture d ecran 2016-02-26 a 10 02 38](https://cloud.githubusercontent.com/assets/505230/13340008/369df232-dc70-11e5-95d4-b14ca830c04d.png)

**Comment by karlcow** [comment]
> For FF, if I remove our hack to make it look good in versions 4-35, it looks like it's still showing the custom arrow even though it's display: none and visibility is tucked behind @supports. I had to add this to hide the arrow - why would older FF be passing @supports ( -moz-appearance: none ) to show the arrow?

Could you point me to a specific example page?

**Comment by toddparker** [comment]
@karlcow - thanks for that suggested fix, good idea. If you open the jsbin and delete the FF hacks at the end AND the little temp fix style I included above (it's in the @support block), the custom  arrow will appear in FF 34 and lower but it shouldn't (you see both the native and custom arrow). The goal is to only show the custom arrow if -moz-appearance is supported and it's odd that it's picking up this rule when it's not supported.

**Comment by karlcow** [comment]
`-moz-appearance` is supported since Firefox 1, so I guess it's normal that `@supports (-moz-appearance: none) {}` is working in Firefox 34.

Here a [test in Firefox 33 on Codepen](http://codepen.io/anon/pen/aNoPVd)

![capture d ecran 2016-02-26 a 14 36 09](https://cloud.githubusercontent.com/assets/505230/13344122/61cfbdde-dc96-11e5-9acf-969604018eb5.png)

Or is there something I didn't get in the example? :stuck_out_tongue_winking_eye: Things can get complicated sometimes.

**Comment by zachleat** [comment]
I’m guessing this issue wasn’t closed when 1.0.3 was released on March 1? Can anyone else confirm?

**Comment by zachleat** [comment]
Closing this, it’s an old one. The work was included with this stuff from last year:
![image](https://user-images.githubusercontent.com/39355/27607974-5add4a8e-5b4b-11e7-9067-57688e9d67ea.png)
